File: num49.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Arrays;

class num49{
    public static void main(String[] args) {
        String[] strs = {"eat", "tea", "tan", "ate", "nat", "bat"};
        List<List<String>> list = num49.groupAnagrams(strs);
        for(List strlist:list){
            for(Object s:strlist){
                System.out.print(s+" ");
            }
            System.out.println('\n');
        }
    }
    public static List<List<String>> groupAnagrams(String[] strs) {
        Map<String,List<String>> map = new HashMap<String,List<String>>();
        for(int i=0; i<strs.length; i++){
            char[] ca = strs[i].toCharArray();
            Arrays.sort(ca);
            String key = String.valueOf(ca);
            if(!map.containsKey(key)){
                map.put(key, new ArrayList<>());
            }
            map.get(key).add(strs[i]);
        }
        return new ArrayList(map.values());
    }
}

File: leetcode/num49.go
package main
import(
	"fmt"
	"strings"
)
func groupAnagrams(strs []string) [][]string {
	var result [][]string
	if len(strs) == 0{
		return result
	}
	var tmp []string
    for i:=1; i<len(strs); i++{
		
	}
	return result
}
func main(){
	strs := []string{"eat", "tea", "tan", "ate", "nat", "bat"}
	fmt.Println(groupAnagrams(strs))
}